### DISCUSSION TITLE
Chronos-2 Post-hoc Explainability

### DISCUSSION THREAD

**Discussion by shihnlee** [amazon-science/chronos-forecasting#492] [Q&A]
Hello,

I am forecasting solar power (output and/or forecast errors) and want to understand covariate contributions.
The setup is roughly:
About 6 months of historical data
Forecast horizon: 24 to 72 hours ahead (48 to 144 steps)
Covariates: weather variables such as solar irradiance (past-only) and weather forecasts plus calendar features like hour of day and day of week (known-future)
Zero-shot inference first, possibly fine-tuning later
1) Has anyone tried applying model-agnostic SHAP (KernelExplainer or PermutationExplainer) to Chronos-2 predictions, treating covariates as the features?  Any caveats or example code to share?
2) How about using SageMaker Debugger or SageMaker Clarify, should I use that?

Thank you.
regards, Shih Nyien

**Comment by daidahao** [comment]
@shihnlee 

Hi, I am not aware of explainability from the official chronos-forecasting package currently.

However, [Chronos-2](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.chronos2_model.html) is also available from [Darts](https://github.com/unit8co/darts) (thanks to @abdulfatir review!). Recently, I have contributed a SHAP explainability feature `TorchExplainer` for any torch-based models in Darts which includes Chronos-2 as well. I use Permutation Explainer as default because it is more efficient than Kernel, but unlike DeepLift and Integrated Gradients, provides base values which many SHAP plots require. See https://github.com/unit8co/darts/pull/3049 for the ongoing PR.

However, the PR is currently under review and has not been merged into main branch. You may try it out by cloning my `daidahao:feature/shap-torch` branch and installing Darts locally. Please us leave feedback if that helps you explain Chronos-2 or other models.

*I have some success with explaining small Chronos-2 on Apple M4 Pro, but obviously your mileage may vary depending on your spec. Permutation is not native PyTorch explainer after all, so NumPy-PyTorch conversions occur under the hood of `TorchExplainer` and cause slowdowns.

**Reply by shihnlee** [reply]
@daidahao 
Thank you for sharing the great work that you have done.
I will spend some time to look into that.

**Comment by shchur** [comment]
@shihnlee Chronos-2 is available in AutoGluon, which provides a simple API to compute feature importances using a permutation-based test. Here is a tutorial showing how this can be done https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-chronos.html#forecasting-with-covariates

**Reply by shihnlee** [reply]
@shchur 
Thank you for sharing.
I will try out the tutorials.
